Warning when building with fast_double_parser

I got the following warning when building with the lib:

fast_double_parser.h(211,1): warning C4715: 'fast_double_parser::leading_zeroes': not all control paths return a value

and looking at the code I believe the warning is correct

--- a/include/fast_double_parser.h
+++ b/include/fast_double_parser.h
@@ -191,7 +191,7 @@ really_inline value128 full_multiplication(uint64_t value1, uint64_t value2) {
 }
 
 /* result might be undefined when input_num is zero */
-int leading_zeroes(uint64_t input_num) {
+inline int leading_zeroes(uint64_t input_num) {
 #ifdef _MSC_VER
   unsigned long leading_zero = 0;
   // Search the mask data from most significant bit (MSB)

--- a/include/fast_double_parser.h
+++ b/include/fast_double_parser.h
@@ -191,19 +191,25 @@ really_inline value128 full_multiplication(uint64_t value1, uint64_t value2) {
 }
 
 /* result might be undefined when input_num is zero */
-inline int leading_zeroes(uint64_t input_num) {
+int leading_zeroes(uint64_t input_num) {
 #ifdef _MSC_VER
   unsigned long leading_zero = 0;
   // Search the mask data from most significant bit (MSB)
   // to least significant bit (LSB) for a set bit (1).
 #ifdef _WIN64
-  if (_BitScanReverse64(&leading_zero, input_num))
-    return (int)(63 - leading_zero);
+  // To avoid warnings, let us force a return. Note that
+  // input_num is non-zero by assumption.
+  (void)_BitScanReverse64(&leading_zero, input_num);
+  return (int)(63 - leading_zero);
 #else
-  if (_BitScanReverse(&leading_zero, (uint32_t)(input_num >> 32)))
+  // To avoid warnings, let us force a return. Note that
+  // input_num is non-zero by assumption.
+  if (input_num >> 32) {
+   (void)_BitScanReverse(&leading_zero, (uint32_t)(input_num >> 32));
     return (int)(63 - (leading_zero + 32));
-  if (_BitScanReverse(&leading_zero, (uint32_t)input_num))
-    return (int)(63 - leading_zero);
+  }
+  (void)_BitScanReverse(&leading_zero, (uint32_t)input_num);
+  return (int)(63 - leading_zero);
 #endif // _WIN64
 #else
   return __builtin_clzll(input_num);

